hyperspy/signal_tools.py

```diff
@@ -960,6 +959,7 @@ def update_plot(self):
         self.reset_span_selector()
         self.update_spectrum_line()
         if len(self.coordinates) > 1:
+            self.signal._plot.pointer.indices = self.coordinates[self.index]
```

Comment:
This is actually a workaround for what may be a minor bug: when ``_auto_update_plot`` is switched from ``False`` to ``True`` the pointer's indices should be set  to the current ones and the patch redrawn, probably around https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/drawing/mpl_hse.py#L55

Reply:
Yes, `auto_update_plot=True` will update the pointer position and redraw the pointer nicely but it will do it for each index and slows down significantly finding spikes, since a indices change of the axes manager will redraw the patch position! So it seems that it is working as expected but not what we want.
`_update_patch_position()` is also working fine but since the actual pointer indices haven't changed, the same thing is redrawn. The only way that I see here is setting the pointer indices manually and changing `_update_patch_position()` to `_on_navigate(self.signal.axes_manager)` does the job here. Actually, just using the pointer position or indices properties would also do, since it trigger `_pos_changed` event and call `_update_patch_position`.